### PR TITLE
Allow installation via pip.

### PR DESCRIPTION
**Pull request by btimby**
Without this change, when pip imports/executes `setup.py` and dependencies are not preinstalled, installation of `ceterach` fails.

This change allows setup.py to run even if dependencies are not yet present. Then pip can install the dependencies and `ceterach`.

### CODE REVIEW

**Comment by btimby** [comment]
Here is an example of the failure.

```python
Collecting ceterach from git+https://github.com/Riamse/ceterach.git#egg=ceterach (from -r /requirements/base.txt (line 113))
  Cloning https://github.com/Riamse/ceterach.git to /tmp/pip-build-0yl8vy9v/ceterach
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-0yl8vy9v/ceterach/setup.py", line 25, in <module>
        from ceterach import __version__
      File "/tmp/pip-build-0yl8vy9v/ceterach/ceterach/__init__.py", line 20, in <module>
        from . import api, page, category, file, revision, user, exceptions
      File "/tmp/pip-build-0yl8vy9v/ceterach/ceterach/api.py", line 27, in <module>
        import requests
    ImportError: No module named 'requests'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-0yl8vy9v/ceterach/
```

**Comment by btimby** [comment]
And then with this change:

```
(venv) btimby@machupo:~/Code/ceterach$ pip install .
Processing /home/btimby/Code/ceterach
Collecting requests>=1.0.3 (from ceterach==0.0.1)
  Using cached requests-2.18.4-py2.py3-none-any.whl
Collecting arrow>=0.3.5 (from ceterach==0.0.1)
Requirement already satisfied: certifi>=2017.4.17 in /home/btimby/Code/wikipedia/venv/lib/python3.5/site-packages (from requests>=1.0.3->ceterach==0.0.1)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /home/btimby/Code/wikipedia/venv/lib/python3.5/site-packages (from requests>=1.0.3->ceterach==0.0.1)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /home/btimby/Code/wikipedia/venv/lib/python3.5/site-packages (from requests>=1.0.3->ceterach==0.0.1)
Requirement already satisfied: idna<2.7,>=2.5 in /home/btimby/Code/wikipedia/venv/lib/python3.5/site-packages (from requests>=1.0.3->ceterach==0.0.1)
Requirement already satisfied: python-dateutil in /home/btimby/Code/wikipedia/venv/lib/python3.5/site-packages (from arrow>=0.3.5->ceterach==0.0.1)
Requirement already satisfied: six>=1.5 in /home/btimby/Code/wikipedia/venv/lib/python3.5/site-packages (from python-dateutil->arrow>=0.3.5->ceterach==0.0.1)
Installing collected packages: requests, arrow, ceterach
  Running setup.py install for ceterach ... done
Successfully installed arrow-0.12.0 ceterach-0.0.1 requests-2.18.4
```

**Comment by Riamse** [comment]
thanks for the PR, will look later.

File: setup.py [hunk 1]
```diff
@@ -22,7 +22,12 @@
 except ImportError:
     from distutils.core import setup
 
-from ceterach import __version__
+# Load __version__ and __author__ from __init__.py without executing the entire
+# file.
+with open('ceterach/__init__.py', 'r') as f:
+    for l in f.readlines():
+        if l[:12] in ('__version__ ', '__author__ ='):
+            exec(l)
```
Comment (Riamse) [inline setup.py]: uh. can we try to think of a better way to do this

**Pull request closed without merging**